### ISSUE TITLE
node-fetch will get stuck (never rejects or resolves) when doing GET with FormData and server redirects

### ISSUE THREAD

**Issue by dzek69** [bitinn/node-fetch#394]
Assuming server under this url returns `Location` header this code will get stuck and never get to the console.log:

```javascript
const formData = new FormData();
const x = await fetch("http://127.0.0.1/admin/redirect.php", {
    body: formData,
}).catch(e => null);
console.log("Done?");
```

I know I should probably not send `body` while doing `GET`, but still `node-fetch` should reject or silently ignore the `body`, just like it do when no redirection happens.

**Comment by bitinn** [comment]
Put `console.log` inside `.then` and `.catch` and see what you got.

**Comment by dzek69** [comment]
What's the difference?
Anyway I did that. Got nothing.

Here's full featured code that reproduces the issue:
```javascript
const FormData = require("form-data");
const fetch = require("node-fetch");

const express = require("express");
const app = express();
app.get("/test", (req, res) => {
    res.redirect("/result");
});
app.get("/result", (req, res) => {
    res.send("ok")
});
app.listen("4356", () => {
    console.log("Listening");
    const formData = new FormData();
    fetch("http://127.0.0.1:4356/test", {
        body: formData, // comment this out if you don't want fetch to stuck
    }).then(() => {
        console.log("aaa");
    }).catch(() => {
        console.log("bbb");
    });
});
```

**Comment by TimothyGu** [comment]
On master (soon to be released as v2.0.0), the test case prints "bbb" with the following error:

```
TypeError: Request with GET/HEAD method cannot have body
    at new Request (/home/timothy-gu/dev/fetch/node-fetch/lib/index.js:1175:10)
    at /home/timothy-gu/dev/fetch/node-fetch/lib/index.js:1351:19
    at new Promise (<anonymous>)
    at fetch (/home/timothy-gu/dev/fetch/node-fetch/lib/index.js:1349:9)
```

I'll consider this fixed.
